backend/config/config_template.py

```diff
@@ -0,0 +1,172 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import dataclasses
+from typing import Dict
+
+import yaml
+
+
+@dataclasses.dataclass
+class DockerConfig:
+    registry: str = "k3d-myregistry.localhost:12345"
+    externalRegistry: str = "docker.io"
+
+
+@dataclasses.dataclass
+class K8sPodSecurityContext:
+    runAsUser: int = 1004370000
+    runAsGroup: int = 1004370000
+    fsGroup: int = 1004370000
+    runAsNonRoot: bool = True
+
+
+@dataclasses.dataclass
+class K8sClusterConfig:
+    imagePullPolicy: str = "Always"
+    podSecurityContext: K8sPodSecurityContext = K8sPodSecurityContext
+
+
+@dataclasses.dataclass
+class K8sPromtailConfig:
+    lokiEnabled: bool = True
+    lokiUrl: str = "http://localhost:30001/loki/api/v1/push"
+    lokiUsername: str = "localLokiUser"
+    lokiPassword: str = "localLokiPassword"
+    serverPort: int = 3101
+
+
+# Only required when using operator k8s
+@dataclasses.dataclass
+class K8sConfig:
+    # Only required, if you'd like to use a local k3d environment
+    context: str = "k3d-collab-cluster"
+    namespace: str = "collab-sessions"
+    storageClassName: str = "local-path"
+    storageAccessMode: str = "ReadWriteOnce"
+    cluster: K8sClusterConfig = K8sClusterConfig
+    promtail: K8sPromtailConfig = K8sPromtailConfig
+    # Only required when no kubectl context is available
+    apiURL: str | None = None
+    token: str | None = None
+
+
+@dataclasses.dataclass
+class GeneralConfig:
+    host: str = "localhost"
+    port: int = 8000
+    scheme: str = "http"
+    wildcardHost: bool = False
+
+
+@dataclasses.dataclass
+class ExtensionGuacamoleConfig:
+    baseURI: str = "http://localhost:8080/guacamole"
+    publicURI: str = "http://localhost:8080/guacamole"
+    username: str = "guacadmin"
+    password: str = "guacadmin"
+
+
+@dataclasses.dataclass
+class ExtensionJupyterConfig:
+    publicURI: str = "http://localhost:8080/jupyter"
+
+
+@dataclasses.dataclass
+class ExtensionsConfig:
+    guacamole: ExtensionGuacamoleConfig = ExtensionGuacamoleConfig
+    jupyter: ExtensionJupyterConfig = ExtensionJupyterConfig
+
+
+@dataclasses.dataclass
+class AuthOauthClientConfig:
+    id: str = "default"
+    secret: str | None = None
+
+
+# Only required when using provider oauth
+@dataclasses.dataclass
+class AuthOathEndpointsConfig:
+    wellknown: str = (
+        "http://localhost:8083/default/.well-known/openid-configuration"
+    )
+    tokenIssuance: str | None = None
+    authorization: str | None = None
+
+
+@dataclasses.dataclass
+class AuthOauthConfig:
+    # Only required when using provider oauth
+    endpoints: AuthOathEndpointsConfig | None = None
+    audience: str = "default"
+    scopes: list[str] = dataclasses.field(default=lambda: ["openid"])
+    client: AuthOauthClientConfig = AuthOauthClientConfig
+    redirectURI: str = "http://localhost:4200/oauth2/callback"
+
+
+@dataclasses.dataclass
+class AuthenticationConfig:
+    provider: str = "oauth"  # oauth | azure
```

Comment:
Support for the Azure provider seems to be missing, check the config_schema for all available option. The validation behaviour shouldn't change.